File: Pi5/src/main/java/org/example/pi5/entities/GameTrade.java
package org.example.pi5.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class GameTrade {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String position; // "short" or "long"

    private int shares;

    private double price;

    private LocalDateTime tradeDate;

    private String state;
    private Double stopLoss;
    private Double takeProfit;

    @ManyToOne
    private GamePortfolio portfolio;

    @ManyToOne
    private Company company;
}

File: materialm-angular-free-main/main/src/app/Models/GameTrade.ts
import { Company } from "./Company";
import { GamePortfolio } from "./GamePortfolio";

export class Gametrade {
    id: number;
    position: string;
    shares: number;
    price: number;
    tradeDate: Date | string;
    portfolio: GamePortfolio;
    stopLoss?: number;
    takeProfit?: number;
    company: Company;
    state: string;
    latestPrice?: number;
}